x-pack/plugins/lens/public/xy_visualization/visual_options_popover/fill_opacity_option.tsx

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFormRow, EuiRange } from '@elastic/eui';
+
+export interface FillOpacityOptionProps {
+  /**
+   * Currently selected value
+   */
+  value: number;
+  /**
+   * Callback on display option change
+   */
+  onChange: (value: number) => void;
+  /**
+   * Flag for rendering or not the component
+   */
+  isFillOpacityEnabled?: boolean;
+}
+
+export const FillOpacityOption: React.FC<FillOpacityOptionProps> = ({
+  onChange,
+  value,
+  isFillOpacityEnabled = true,
+}) => {
+  return isFillOpacityEnabled ? (
```

Comment:
Could we use `x-pack/plugins/lens/public/shared_components/debounced_value.ts` here? It's a flexible hook to debounce things like this so the chart doesn't re-render too often.

Reply:
Nice! Sure